rules/prefer-await.js

```diff
@@ -0,0 +1,210 @@
+'use strict';
+const {methodCallSelector} = require('./selectors/index.js');
+const {removeParentheses, removeMethodCall} = require('./fix/index.js');
+const {getParenthesizedText, isParenthesized} = require('./utils/parentheses.js');
+const shouldAddParenthesesToCallExpressionCallee = require('./utils/should-add-parentheses-to-call-expression-callee.js');
+const getFunctionParameterVariables = require('./utils/get-function-parameter-variables.js');
+const needsSemicolon = require('./utils/needs-semicolon.js');
+
+const MESSAGE_ID_ERROR = 'prefer-await/error';
+const MESSAGE_ID_SUGGESTION = 'prefer-await/suggestion';
+const messages = {
+	[MESSAGE_ID_ERROR]: 'Do not use `Promise#{{method}}(…)`.',
+	[MESSAGE_ID_SUGGESTION]: 'Use `await` expression (UNSAFE).',
+};
+
+function isInsideTryStatement(node, stopNode) {
```

Comment:
I'm thinking we could make a more general utility for this. Something like `hasAncestor(node, {stopNode})`. It could potentially use another utility internally: `traverseUp()` which would be an Iterable that simply traverses parents.

If you agree, but don't have time to do it now, just add it as a TODO comment.